src/index.js

```diff
@@ -1,113 +1,109 @@
-
+function getConvertingFunctionByType(type) {
+  let convertingFunc;
+  switch (type) {
+    case Types.Number:
+      convertingFunc = Number;
+      break;
+    case Types.Boolean:
+      convertingFunc = (v) => v === 'true';
+      break;
+    case Types.String:
+      convertingFunc = String;
+      break;
+    case Types.Array:
+      convertingFunc = (v) => v.split(',');
+      break;
+    default:
+      convertingFunc = (v) => v;
+  }
+
+  return convertingFunc;
 }
 
-function convertValueToRequiredType (type, value) {
-    const convertingFunction = getConvertingFunctionByType(type)
-    return convertingFunction(value)
+function convertValueToRequiredType(type, value) {
+  const convertingFunction = getConvertingFunctionByType(type);
+  return convertingFunction(value);
 }
 
-function createEnvObject (paramsConfig, shouldValidateEnvParams = true) {
-    if (shouldValidateEnvParams) {
-        validateMandatoryEnvParams(paramsConfig)
-    }
-    const result = {}
-    Object.entries(paramsConfig)
-        .forEach(([paramName, config]) => {
-            let value = process.env[paramName] || config.default
-
-            if (config.wrappingFunction || config.type) {
-                if (config.type) {
-                    value = convertValueToRequiredType(config.type, value)
-                } else {
-                    value = applyWrappingFunction(paramName, config.wrappingFunction, value)
-                }
-
-            } else {
-                const wrappingFunction = getWrappingFunctionByDefaultValue(config.default)
-                value = applyWrappingFunction(paramName, wrappingFunction, value)
-            }
-
-            if (shouldValidateEnvParams) {
-                if (config.validationFunction) {
-                    applyValidationFunction(paramName, config, value)
-                }
-
-                if (config.validValues) {
-                    if (!config.validValues.includes(value)) {
-                        throw new Error(`value '${value}' of env param '${paramName}' is not valid, check paramsConfig to see valid values`)
-                    }
-                }
-            }
-
-            result[paramName] = value
-        })
-
-    return result
+function createEnvObject(paramsConfig, shouldValidateEnvParams = true) {
+  if (shouldValidateEnvParams) {
+    validateMandatoryEnvParams(paramsConfig);
+  }
+  const result = {};
+  Object.entries(paramsConfig)
+    .forEach(([paramName, config]) => {
+      let value = process.env[paramName] || config.default;
+
+      if (config.type) {
+        value = convertValueToRequiredType(config.type, value);
+      } else if (config.wrappingFunction) {
+        value = applyWrappingFunction(paramName, config.wrappingFunction, value);
+      } else {
+        const wrappingFunction = getWrappingFunctionByDefaultValue(config.default);
+        value = applyWrappingFunction(paramName, wrappingFunction, value);
+      }
+
+      if (shouldValidateEnvParams) {
```

Comment:
`if (shouldValidateEnvParams && config.validationFunction)`

Reply:
why? i want `      if (shouldValidateEnvParams) { ` to wrap both cases:
       ` if (config.validationFunction) {`
     and  ` if (config.validValues && !config.validValues.includes(value)) {`

avoiding nested `if`'s here will result in appending `shouldValidateEnvParams &&` to both cases which is not readable IMO

Reply:
ok